### ISSUE TITLE
Add gradient norm clipping to ``TitanTrainer``

### ISSUE THREAD

**Issue by joecummings** [meta-pytorch/forge#253]
### Context

Even though it's possible to do gradient norm with our integration with TorchTitan, we do not enable it yet in ``TitanTrainer``. Gradient norm clipping is a common practice to increase the stability of training.

### To do

Add the following code example of gradient clipping [here](https://github.com/pytorch/torchtitan/blob/1b9cfda0c981b36661bf7ba70b4581676c8f6a76/torchtitan/experiments/forge/example_train.py#L242) into the ``TitanTrainer`` [after forward_backward but before taking an optimizer step](https://github.com/meta-pytorch/torchforge/blob/47c2333e87ccd8731cd984719e9959cf3d3efd3e/src/forge/actors/trainer/titan.py#L166). This should be controllable through the config ``trainer.grad_norm: 1.0``.  

### Acceptance criteria 

In order to properly test this, you will need to log the grad norm in addition to the loss. Before the change, grad norm should be *anything* and after the change, it should be shown to be clipped to a max of X. Any PR addressing this change should show WandB logs to this effect.

### Anything else?

Handling AMP in this case is out of the scope for this task.

**Comment by krammnic** [comment]
Let me take this, but we probably need to separate this on 2 flows (with and without AMP) to make clipping correct:

https://github.com/meta-pytorch/forge/blob/94557d1dcf364c1e9e0fc64b4e32dcfdadde9d2d/src/forge/actors/trainer.py#L237

We should do this in order to correctly respect `GradScaler` in combination with clipping. Correct me if I'm wrong.
